src/status_im/ui/screens/chat/views.cljs

```diff
@@ -109,9 +110,48 @@
 
 (defn empty-chat-container
   []
-  [react/view style/empty-chat-container
-   [react/text {:style style/empty-chat-text}
-    (i18n/label :t/empty-chat-description)]])
+  (let [{:keys [color chat-id chat-name]}
+        @(re-frame/subscribe [:chats/current-chat])
+
+        link
+        (universal-links/generate-link :public-chat :external chat-id)
```

Comment:
computations should be done in subscriptions

Reply:
I see. I understand why computations should be done in subscriptions but not sure about which subscription. Like in `:chats/current-chat`? or new subscription?

Can you please elaborate?

Reply:
it could be calculated and stored in app-db at the time of chat creation right? But I kinda followed what was done in chat actions which also has share chat item.

https://github.com/status-im/status-react/blob/aa0c45738200d64b39f95955b2e518740e3080e1/src/status_im/ui/screens/chat/actions.cljs#L15

I could move the calculation to private function like it does.

Reply:
did it as in chat actions
done!

Reply:
Ideally in :chats/current-chat subscription if there is no messages (chat empty), you add a compute and add the link key, no need to persist it to app-db. Chat actions code isn't a good example it should be refactored at some point

Reply:
ok got it!
another update coming

Reply:
done!